### ISSUE TITLE
Multiple mappings on a field does not work for nested docs

### ISSUE THREAD

**Issue by pirovorster** [elastic/elasticsearch#20886]
**Elasticsearch version**: 1.74

**Plugins installed**: []

**JVM version**: Found Hosting 

**OS version**: Found Hosting

I followed https://www.elastic.co/guide/en/elasticsearch/guide/current/mixed-lang-fields.html to analyze fields multiple times. However it seems as if this feature does not work on nested documents.

I added an index with the following mapping:
```
{
  "mappings": {
    "case": {
      "properties": {
        "clientRelationship.policy.policyHolder.firstName": {
          "type": "string",
          "fields": {
            "key": {
              "type": "string",
              "analyzer": "keyword"
            }
          }
        },
        "test": {
          "type": "string",
          "fields": {
            "key": {
              "type": "string",
              "analyzer": "keyword"
            }
          }
        },
        "test2": {
          "type": "string",
          "analyzer": "keyword"
        }
      }
    }
  }
}
```
Then added a document:
```
{
  "clientRelationship": {
    "policy": {
      "policyHolder": {
        "firstName": "Jeff Awesome"
      }
    }
  },
  "test": "Jeff Awesome"
}
```
When searching
```
{
  "query": {
    "term": {
      "test": "Jeff Awesome"
    }
  }
}
```
returns a document, while
```
{
  "query": {
    "term": {
      "clientRelationship.policy.policyHolder.firstName.key": "Jeff Awesome"
    }
  }
}
```
does not

**Comment by jpountz** [comment]
I think you are hitting an issue due to the fact that you put dots in field names in your mappings. Can you try with the following mapping?

```
{
  "mappings": {
    "case": {
      "properties": {
        "client": {
          "properties": {
            "policy": {
              "properties": {
                "policyHolder": {
                  "properties": {
                    "firstName": {
                      "type": "string",
                      "fields": {
                        "key": {
                          "type": "string",
                          "analyzer": "keyword"
                        }
                      }
                    }
                  }
                }
              }
            }
          }
        },
        "test": {
          "type": "string",
          "fields": {
            "key": {
              "type": "string",
              "analyzer": "keyword"
            }
          }
        },
        "test2": {
          "type": "string",
          "analyzer": "keyword"
        }
      }
    }
  }
}
```

**Comment by pirovorster** [comment]
Thanks so much. That worked like a charm.

**Comment by jpountz** [comment]
Thanks for confirming. This problem is fixed in 5.0 ( the initial example you gave would work) so I will close this issue.
